lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -417,6 +515,72 @@ void ScienceSensorsSystemPrivate::ReadData()
   this->initialized = true;
 }
 
+/////////////////////////////////////////////////
+void ScienceSensorsSystemPrivate::UpdateWorldSphericalOrigin(
+  ignition::gazebo::EntityComponentManager &_ecm)
+{
+  // Lock for changes to worldOriginSpherical* until update complete
+  std::lock_guard<std::mutex> lock(mtx);
+
+  if (!this->worldSphericalCoordsInitialized)
+  {
+    auto latLon = this->world.SphericalCoordinates(_ecm);
+    if (latLon)
+    {
+      ignwarn << "New spherical coordinates detected: "
+          << latLon.value().LatitudeReference().Degree() << ", "
+          << latLon.value().LongitudeReference().Degree() << std::endl;
+
+      this->worldOriginSphericalCoords = latLon.value();
+
+      // Extract world origin in (lat, long, elevation) from spherical coords
+      this->worldOriginSphericalPos = ignition::math::Vector3d(
+        this->worldOriginSphericalCoords.LatitudeReference().Degree(),
+        this->worldOriginSphericalCoords.LongitudeReference().Degree(),
+        0);
+      // Convert spherical coordinates to Cartesian
+      this->sphCoord = ignition::math::SphericalCoordinates(
+        this->worldOriginSphericalCoords.Surface());
+      this->worldOriginCartesianCoords =
+        this->sphCoord.LocalFromSphericalPosition(
+          this->worldOriginSphericalPos);
+
+      this->worldSphericalCoordsInitialized = true;
+
+      //TODO(chapulina) Shift science data to new coordinates
```

Comment:
I think this TODO can be moved into `ShiftDataToNewSphericalOrigin`

Reply:
Because of this https://github.com/osrf/lrauv/pull/70#discussion_r755672172 , I will leave this here, because this `if` logic may need to be changed.